### DISCUSSION TITLE
nuxt/kit component lib composables not found in nuxt3

### DISCUSSION THREAD

**Discussion by zerounix** [nuxt/nuxt#17306] [Questions]
Hi guys,

i'm banging my head now for a few days and want to ask for support. I'm migrating my component lib from vue2 to vue3 and updated my nuxt project to nuxt3. I used unplugin-auto-import and unplugin-vue-components, which is really cool stuff, all over my project. 
When it now comes to integration in my nuxt project i used this way: https://v3.nuxtjs.org/guide/directory-structure/components#library-authors . So i come up with a nuxt.ts in my compnent library auto-importing my composables and adding my components directory:

```
import { fileURLToPath } from 'url'
import { addComponentsDir, defineNuxtModule, createResolver, addAutoImport } from '@nuxt/kit'
import consola from 'consola'

const logger = consola.create({})

const rPath = (p: string) => fileURLToPath(new URL(p, import.meta.url).toString())

export default defineNuxtModule({
meta: {
    name: '@complib/ui-next',
  },
  async setup(options, nuxt) {
   ...
    const { resolve } = createResolver(import.meta.url)

    
    // register components
    addComponentsDir({ path: rPath('./components'), prefix: '', pathPrefix: false })
    logger.success(".. added components")

    // register composables
    nuxt.hook('autoImports:dirs', (dirs) => {
      dirs.push(resolve(rPath('./composables')))
    })
   
   ... 
  }
})
```

In final bundle i have a dist/components and a dist/composable folder. I can find my components in components.d.ts and my composables in imports.d.ts in .nuxt folder. So the module seems to work. But when i start the nuxt server and my component lib boots up it can't find my composable. 

The error looks like this:

```
[nuxt] [request error] useXYZ is not defined
  at setup (./.nuxt/dist/server/server.mjs:8209:24) 
```

but .nuxt/imports.d.ts has it

```
export { useXYZ } from '/Users/user/Develop/next/portal/node_modules/@complib/ui-next/dist/composables/useXYZ/index';

```
I searched the @nuxt/kit issues and found this ticket https://github.com/nuxt/framework/issues/5158 which happend also to me, but my workaround was to rebuild everything with <script lang="ts" setup> which got rid of using defineComponent. 

nuxt and @nuxt/kit version are "@nuxt/kit": "^3.0.0-rc.4", and "nuxt": "3.0.0-rc.4"

Can anybody please advise me what i'm doing wrong?

**Comment by Diizzayy** [comment]
@zerounix are you using [Nuxt Module Builder](https://github.com/nuxt/module-builder#srcruntime) to build the nuxt module? If so, your composables and components [need to be in the `src/runtime` directory as mentioned](https://github.com/nuxt/module-builder#srcruntime) to properly make them available to your app. Otherwise if you're directly using [unbuild](https://github.com/unjs/unbuild) for example, you'll need to manually add your pertinent code (composables / components) entries to be transpiled.

**Reply by zerounix** [reply]
@Diizzayy no im using plain old vite to bundle my lib, and copy over folders with rollup copy plugin. I tried unbundle but could'nt get it to work for some reason.

How can i add my code for transpiling? Sorry if that question is stupid :)

**Reply by zerounix** [reply]
I tried  build.transpile in nuxt.config.ts and nuxt.options.build.transpile in my nuxt.ts in comoponent lib. Both did not work.
